lms/djangoapps/mobile_api/course_info/views.py

```diff
@@ -321,10 +321,13 @@ def list(self, request, **kwargs):  # pylint: disable=W0221
             request - Django request object
         """
 
-        response = super().list(request, kwargs)
+        api_version = self.kwargs.get('api_version')
+        if api_version == 'v4':
+            response = super().list(request)
+        else:
+            response = super().list(request, kwargs)
```

Comment:
1. Put yourself in the shoes of a developer who comes across this code in the future. Would they have any idea what is going on here? Probably not. Please leave a comment explaining why you're doing this, and why it only applies to v4 and not other API versions.
2. Do you really need to hide the entire kwargs list from the superclass, or is it just a certain kwarg that you need to hold back? If it is just a certain kwarg, then it would be more self-explanatory and also less bug-prone to just remove that one kwarg.